pageserver/src/tenant/remote_timeline_client/download.rs

```diff
@@ -428,14 +418,7 @@ pub(crate) async fn download_initdb_tar_zst(
             let mut download = tokio_util::io::StreamReader::new(download.download_stream);
             let mut writer = tokio::io::BufWriter::with_capacity(super::BUFFER_SIZE, file);
 
-            // TODO: this consumption of the response body should be subject to timeout + cancellation, but
-            // not without thinking carefully about how to recover safely from cancelling a write to
-            // local storage (e.g. by writing into a temp file as we do in download_layer)
```

Comment:
Why is this comment no longe relevant?

Reply:
In #6697 I moved the timeout into remote_storage but neglected to think about the external enough (why this PR is needed).